From 5e8ffbce4ff82a6e8045d64a788bed2222bfb376 Mon Sep 17 00:00:00 2001
From: NIIBE Yutaka <gniibe@fsij.org>
Date: Tue, 21 Aug 2018 18:54:24 +0900
Subject: [PATCH 2/4] jtag: bbg-swd: Fix bbg_swd_interface_init and
 bbg_swd_execute_reset.

* src/jtag/drivers/bbg-swd.c (bbg_swd_interface_init): Call
bbg_swd_switch_seq with LINE_RESET.
(bbg_swd_execute_reset): Add delay.
---
 src/jtag/drivers/bbg-swd.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/jtag/drivers/bbg-swd.c b/src/jtag/drivers/bbg-swd.c
index 2a4fa812..19548688 100644
--- a/src/jtag/drivers/bbg-swd.c
+++ b/src/jtag/drivers/bbg-swd.c
@@ -1,5 +1,5 @@
 /***************************************************************************
- *   Copyright (C) 2016  Flying Stone Technology                           *
+ *   Copyright (C) 2016, 2018  Flying Stone Technology                     *
  *   Author: NIIBE Yutaka <gniibe@fsij.org>                                *
  *                                                                         *
  *   This program is free software; you can redistribute it and/or modify  *
@@ -126,6 +126,9 @@ static int bbg_swd_interface_init(void)
 		}
 	}
 
+	if (bbg_swd_switch_seq(LINE_RESET) != ERROR_OK)
+		return ERROR_FAIL;
+
 	LOG_INFO("BBG-SWD: Interface ready");
 
 	return ERROR_OK;
@@ -327,6 +330,7 @@ static const struct command_registration bbg_swd_command_handlers[] = {
 static void bbg_swd_execute_reset(struct jtag_command *cmd)
 {
 	bbg_swd_gpio_srst(cmd->cmd.reset->srst ? 0: 1);
+	alive_sleep (10);
 }
 
 static void bbg_swd_execute_sleep(struct jtag_command *cmd)
-- 
2.18.0

